addons/portal/controllers/portal.py

```diff
@@ -185,14 +209,30 @@ def account(self, redirect=None, **post):
             'error_message': [],
         })
 
+        login = None
         if post and request.httprequest.method == 'POST':
+            login = post.pop("login")
             if not partner.can_edit_vat():
                 post['country_id'] = str(partner.country_id.id)
 
             error, error_message = self.details_form_validate(post)
+            if not login:
+                error["login"] = "missing"
+                error_message.append(_("Login ID can not be empty!"))
             values.update({'error': error, 'error_message': error_message})
             values.update(post)
             if not error:
+                if request.env.user.login != login:
+                    try:
+                        with request.env.cr.savepoint():
+                            request.env.user.write({"login": login})
+                        # update session token so the user does not get logged out
+                        request.session.session_token = request.env.user._compute_session_token(request.session.sid)
+                    except (ValidationError, psycopg2.errors.UniqueViolation) as e:
+                        error["login"] = "Duplicate"
+                        error_message.append(_("You can not have two users with the same login."))
+                        values.update({'error': error, 'error_message': error_message})
+                        return self._prepare_my_details_values_and_return(values, partner, redirect, login)
```

Comment:
Not sure why login error deserved an early return but not the other types of errors. If login is popped from the values it should be fine to continue no?